### PR TITLE
List workspaces with most recently running jobs

### PR DESCRIPTION
**Pull request by keyihuang**
## PR status: 
Dev work is done & PR is under review

## Related PR:
https://github.com/airbytehq/airbyte-cloud/pull/4116

## What
This is to enhance the efficiency of `DelinquentWorkspaceCronJob` in `airbyte-cloud/billing`.

## How
Instead of running the `DelinquentWorkspaceCronJob` for all workspaces, now we want to focus on the workspaces with most recently running jobs (e.g. in the last 48 hours)

## Recommended reading order
1. `airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java`
2. test files

## Can this PR be safely reverted / rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java [hunk 1]
```diff
@@ -761,6 +764,29 @@ public List<DestinationConnection> listWorkspaceDestinationConnection(final UUID
...
+        .on(CONNECTION.SOURCE_ID.eq(ACTOR.ID))
+        .join(JOBS)
+        .on(CONNECTION.ID.cast(VARCHAR(255)).eq(JOBS.SCOPE))
+        .where(JOBS.UPDATED_AT.greaterOrEqual(OffsetDateTime.now().minusHours(timeWindowInHours)))
+        .fetch());
+    final List<UUID> workSpaceIds = new ArrayList<>();
+    for (final Record record : records) {
+      workSpaceIds.add(record.get(ACTOR.WORKSPACE_ID));
+    }
+    return workSpaceIds;
```
Comment (xiaohansong) [inline airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java:787]: nit: use a lambda for better readability (follow example on line 803)

**Review by xiaohansong — CHANGES_REQUESTED** [review]
Nice work. Can we add a test on this?

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java (hunk 1 above)
Comment (keyihuang) [reply airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java]: sure, resolved

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspaceFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    configRepository = spy(new ConfigRepository(
+        database,
+        new ActorDefinitionMigrator(new ExceptionWrappingDatabase(database)),
+        null));
+  }
+
+
+  @Test
+  @DisplayName("should return the a list of workspace IDs with most recently running jobs")
+  void testListWorkspacesByMostRecentlyRunningJobs() throws IOException {
```
Comment (xiaohansong) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspaceFilterTest.java:116]: Can you add a comment saying how the tests are setup and why eventually those 3 workspaces were selected, so users do not have to read through all setup code when they need a closer look at this test?

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspaceFilterTest.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+
+  @BeforeEach
+  void beforeEach() {
+    setup();
+  }
+
+  void setup() {
+    configRepository = spy(new ConfigRepository(
```
Comment (xiaohansong) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspaceFilterTest.java:107]: why this needs to be a spy?

**Review by xiaohansong — APPROVED** [review]
This looks great! Leaving two comments for the test and once fixed this is approved

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspaceFilterTest.java (hunk 2 above)
Comment (keyihuang) [reply airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspaceFilterTest.java:114]: sure, added some comments below.

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspaceFilterTest.java (hunk 3 above)
Comment (keyihuang) [reply airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/WorkspaceFilterTest.java]: good catch, removed spy, it works fine as a real object test. spy is not required.

**Pull request merged**